### PR TITLE
Refine docs of `Text` ( add `disable_ligatures=True` for t2c)

### PR DESCRIPTION
**Pull request by kolibril13**
now:
image
before:
"w" was still white, because white spaces were also counted as a character.
After this pr find_indexes can distinguish between cases that need to count white spaces, and those which don't.

### CODE REVIEW

File: manim/mobject/svg/text_mobject.py [hunk 1]
```diff
@@ -322,14 +322,14 @@ def construct(self):
...
-                self.add(text0,text1, text2,text3)
-                for i,mobj in enumerate(self.mobjects):
-                    mobj.shift(DOWN*(i-1))
+                text0 = Text('Hello world', t2c={'world':YELLOW})
+                text1 = Text('Hello world', slant=ITALIC)
+                text2 = Text('Hello world', t2s={'world':ITALIC})
+                text3 = Text('Hello world', weight=BOLD)
+                text4 = Text('Hello world', t2w={'world':BOLD})
 
+                self.add(text0,text1,text2,text3,text4)
```
Comment (Darylgolden) [inline manim/mobject/svg/text_mobject.py:331]:
```suggestion
                self.add(text0, text1, text2, text3, text4)
```

File: manim/mobject/svg/text_mobject.py [hunk 2]
```diff
@@ -557,7 +559,10 @@ def set_color_by_t2c(self, t2c=None):
         """Internally used function. Sets colour for specified strings."""
```
Comment (Darylgolden) [inline manim/mobject/svg/text_mobject.py:559]:
```suggestion
        """Internally used function. Sets color for specified strings."""
```

**Review by Darylgolden — CHANGES_REQUESTED** [review]
LGTM, just minor things

**Comment by naveen521kk** [comment]
@kolibril13 Can you give an example of what this fixes? I mean copy-pasting an example with screenshots of how it rendered before and after.

**Comment by naveen521kk** [comment]
Also, can you try the coloring with `disable_ligature` flag?

**Comment by kolibril13** [comment]
image
before this pr.
you can test it with:
`Text("Visit us at docs.manim.community",t2c={'docs.manim.community':YELLOW}).scale(1.3).shift(DOWN)`
on this branch, this is fixed.

**Comment by naveen521kk** [comment]
It's because of ligatures and not spaces. Can you try disabling ligatures and then try?
```
Text("Visit us at docs.manim.community",t2c={'docs.manim.community':YELLOW}, disable_ligatures=True).scale(1.3).shift(DOWN)
```

**Comment by kolibril13** [comment]
releated to https://github.com/ManimCommunity/manim/issues/776

**Comment by kolibril13** [comment]
First image is from master, second image is from this branch.
So both is working: A) disable_ligatures and B) removing white spaces
image

**Comment by naveen521kk** [comment]
This PR breaks colours even when `disable_ligatures` are on. For example,
```py
from manim import *


class a(Scene):
    def construct(self):
        a = Text(
            "Visit us at fl docs.manim.community",
            disable_ligatures=True,
            t2c={"docs.manim.community": YELLOW},
        )
        b = Text(
            "Visit us at fl docs.manim.community",
            disable_ligatures=False,
            t2c={"docs.manim.community": RED},
        )
        self.add(Group(a, b).arrange(DOWN))
```
render's as
a_ManimCE_v0 6 0

in this branch, while it renders as this in the master branch(which is expected.)
bot
which is expected one

File: manim/mobject/svg/text_mobject.py [hunk 3]
```diff
@@ -323,14 +323,14 @@ def construct(self):
...
-                self.add(text0,text1, text2,text3)
-                for i,mobj in enumerate(self.mobjects):
-                    mobj.shift(DOWN*(i-1))
-
+                text1 = Text('Hello world', slant=ITALIC)
+                text2 = Text('Hello world', t2s={'world':ITALIC})
+                text3 = Text('Hello world', weight=BOLD)
+                text4 = Text('Hello world', t2w={'world':BOLD})
+                text5 = Text('Hello world', t2c={'o':YELLOW}, disable_ligatures=True)
+                text6 = Text("Visit us at docs.manim.community",t2c={'docs.manim.community':YELLOW}, disable_ligatures=True).scale(1.3).shift(DOWN)
```
[the remark this answers is not in the document]
Comment (naveen521kk) [reply manim/mobject/svg/text_mobject.py:331]: Also, chaining methods make the line so long that it's unreadable.

File: manim/mobject/svg/text_mobject.py [hunk 4]
```diff
@@ -323,14 +323,17 @@ def construct(self):
...
-                    mobj.shift(DOWN*(i-1))
-
+                text1 = Text("Hello world", slant=ITALIC)
+                text2 = Text("Hello world", t2s={'world':ITALIC})
+                text3 = Text("Hello world", weight=BOLD)
+                text4 = Text("Hello world", t2w={'world':BOLD})
+                text5 = Text("Hello world", t2c={'o':YELLOW}, disable_ligatures=True)
+                text6 = Text("Visit us at docs.manim.community",
+                    t2c={"docs.manim.community": YELLOW},
+                    disable_ligatures=True)
```
Comment (naveen521kk) [inline manim/mobject/svg/text_mobject.py:333]: Maybe?
```suggestion
                text6 = Text(
                    "Visit us at docs.manim.community",
                    t2c={"docs.manim.community": YELLOW},
                    disable_ligatures=True,
               )
```

**Review by naveen521kk — APPROVED** [review]
sgtm, will merge when ci passes

**Pull request merged**